integration-tests/pkg/mock_sensor/expect_conn.go

```diff
@@ -72,14 +72,90 @@ loop:
 			if conn.GetContainerId() != containerID {
 				continue loop
 			}
-
 			if len(s.Connections(containerID)) == n {
 				return s.Connections(containerID)
 			}
 		}
 	}
 }
 
+// checkIfConnectionsMatchExpected compares a list of expected and observed connection match exactly.
+func (s *MockSensor) checkIfConnectionsMatchExpected(t *testing.T, connections []types.NetworkInfo, expected []types.NetworkInfo) bool {
+	if len(connections) > len(expected) {
+		return assert.ElementsMatch(t, expected, connections, "networking connections do not match")
+	}
+
+	if len(connections) == len(expected) {
+		types.SortConnections(connections)
+		for i := range expected {
+			if !expected[i].Equal(connections[i]) {
+				return assert.ElementsMatch(t, expected, connections, "networking connections do not match")
+			}
+		}
+		return true
+	}
+	return false
+}
+
+// getConnectionsAndCompare gets the connections for a container, sorts them if order is set to true, and compares with a set of expected
+// connections. If asssertMismatch is true and the set of observed connections does not match the set of expected connections an assert is
+// triggered. If assertMismatch is not set then just return if the observed and expected connections match.
+func (s *MockSensor) getConnectionsAndCompare(t *testing.T, containerID string, order bool, assertMismatch bool, expected ...types.NetworkInfo) bool {
```

Comment:
If `checkIfConnectionsMatchExpected` always sorts the container connections, what is the point of ordering them here based on the `order` parameter?

Reply:
Also, why do we need `assertMismatch`? Couldn't the caller just check if the returned value is false?

Reply:
On a separate note about `order`, the way collector handles connections is inherently asynchronous, so even with extreme care, if the implementation preserved the order, expecting a given order of connections to always be in the same order seems unrealistic and a great way to add flakes to our tests.

Reply:
+1 to @Molter73's point about avoiding reliance on connection order. We should probably always sort both lists (from collector and the expected info from the tests themselves)

Reply:
It might even make sense for `s.Connections()` to return a sorted list

Reply:
We may expect a certain order when all of the connections are reported in different scrapes. The connections within a scrape are not ordered, but the scrapes themselves are ordered.

Reply:
> If checkIfConnectionsMatchExpected always sorts the container connections, what is the point of ordering them here based on the order parameter?

That was a mistake. The connections should not be sorted in `checkIfConnectionsMatchExpected`. I have removed that sort.

Reply:
> Also, why do we need assertMismatch? Couldn't the caller just check if the returned value is false?

We don't. I did that to simplify `CompareConnections`.

Reply:
> It might even make sense for s.Connections() to return a sorted list

I don't like this idea as we might want to not sort the connections in the future. However, I have created `SortConnections` which does what you propose.

Reply:
> We may expect a certain order when all of the connections are reported in different scrapes. The connections within a scrape are not ordered, but the scrapes themselves are ordered.

Perhaps this is reason enough to store a full network message ([NetworkConnectionInfo](https://github.com/stackrox/stackrox/blob/master/proto/internalapi/sensor/network_connection_info.proto#L13)) from Collector rather than extracting individual connections or endpoints. That way we can more accurately test behaviour for a given scrape (represented by a single message.) Another reason is that we currently have no way of deterministically waiting for a scrape to happen and the existing tests that do this simply make a best guess, but if we could wait for the mock sensor to receive a network message then we have a very neat way of doing that wait.

At that point the notion of whether we should or shouldn't sort Connections becomes a non-issue.

> I don't like this idea as we might want to not sort the connections in the future.

We can always undo it in the future if it becomes a problem

Reply:
I am now doing the sort in `Connections` and have removed `SortedConnections`.